makefiles/boards/stm32.inc.mk

```
@@ -0,0 +1,54 @@
+PROGRAMMER ?= openocd
+
+ifeq (bmp,$(PROGRAMMER))
+  # On Blackmagic Probe, the first ACM is used to connect to the gdb server,
+  # the second is the BMP's UART interface
+  PORT_LINUX ?= /dev/ttyACM1
+  PORT_DARWIN ?= $(wordlist 2, 2, $(sort $(wildcard /dev/tty.usbmodem*)))
+else
+  # configure the serial terminal
+  PORT_LINUX ?= /dev/ttyACM0
+  PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.usbmodem*)))
+endif
+
+# setup serial terminal
+include $(RIOTMAKE)/tools/serial.inc.mk
+
+ifeq (openocd,$(PROGRAMMER))
+  # STM32 boards can become un-flashable after a hardfault,
+  # use connect_assert_srst to always be able to flash or reset the boards.
+  export OPENOCD_RESET_USE_CONNECT_ASSERT_SRST ?= 1
+
+  # For STM32 boards the ST-link adapter is the default adapter, e.g. all
+  # Nucleo boards have an on-board ST-link adapter
+  DEBUG_ADAPTER ?= stlink
+
+  # RIOT uses openocd by default
+  include $(RIOTMAKE)/tools/openocd.inc.mk
+else
+  ifeq (bmp,$(PROGRAMMER))
+    include $(RIOTMAKE)/tools/bmp.inc.mk
+  else
+    ifeq (dfu-util,$(PROGRAMMER))
```

No need to add another level of nesting, `$(PROGRAMMER)` doesn't change.

```suggestion
ifeq (openocd,$(PROGRAMMER))
  …
endif
ifeq (bmp,$(PROGRAMMER))
  …
endif
ifeq (dfu-util,$(PROGRAMMER))
  …
endif
```

---

A pity that there is no `switch` or `elifeq`

---

Done